locations/descriptor_locations_test.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 		https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package locations
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/jhump/protoreflect/desc"
+)
+
+func TestDescriptorName(t *testing.T) {
+	f := parse(t, `
+		message Foo {
+		  string bar = 1;
```

Comment:
Can you add a map field for testing?

Reply:
Just did. Incidentally, they work fine (it does not return nil).

Reply:
In that case, can you double check the rule `core::0192::has-comments`? Last night, I had some issues running this rule against the `google/cloud/iot/v1/resources.proto`
```
api-linter google/cloud/iot/v1/resources.proto
```
It checked a message `MetadataEntry`, which I believe was generated by the map field `map<string, string> metadata = 17;` in the `Device` message. And since it is generated, there is no comment. So once we fix the `nil` issue here, the rule will complain about the generated message.

Reply:
Sure. Let's make that a separate task though. Can you file an issue?

Reply:
Done. Issue #311

Reply:
Done. Issue #311